src/QMCHamiltonians/NonLocalECPComponent.cpp

```diff
@@ -866,13 +865,9 @@ void NonLocalECPComponent::evaluateOneBodyOpMatrixdRContribution(ParticleSet& W,
 }
 
 ///Randomly rotate sgrid_m
-void NonLocalECPComponent::randomize_grid(RandomGenerator& myRNG)
+void NonLocalECPComponent::randomize_grid(const RandomRotationState& rs)
```

Comment:
Note that there is also NonLocalECPComponent::randomize_grid(std::vector<T>& sphere, RandomGenerator& myRNG). From what I can tell, this is only used by the legacy NonLocalECPotential_CUDA.cpp. For consistency, should that randomize_grid call be changed as well since the legacy code hasn't been removed yet?

That probably breaks more tests and since we are dropping eventually dropping the legacy, maybe it isn't worth changing. But it seems to me things should be done consistently for whatever is actively in the code base.

Reply:
Indeed, we will likely want to fix this everywhere.

The code used by the legacy CUDA code paths will also need to be updated.

There will have to be some discussion around updating the tests. This will be a significant effort even for the smallest possible update and likely need parallelizing over multiple people.